src/components/ArmyBuildPane.jsx

```diff
@@ -17,30 +17,42 @@ const Container = styled.div`
 `;
 
 const ArmyBuildPane = () => {
+
   const [unitList, setUnitList] = useState({});
   const [listIndex, setListIndex] = useState(1);
```

Comment:
Just seeing this for the first time. Any reason you're using a map and a list index to store your units? One weird side effect is that when you remove units, the listIndex isn't adjusted so you end up with a map with missing indices.